metadata-io/src/main/java/com/linkedin/metadata/search/elasticsearch/query/request/SearchQueryBuilder.java

```diff
@@ -89,41 +94,46 @@ private static Set<Pair<String, Float>> getStandardFields(@Nonnull EntitySpec en
 
       String fieldName = fieldSpec.getSearchableAnnotation().getFieldName();
       double boostScore = fieldSpec.getSearchableAnnotation().getBoostScore();
-      fields.add(Pair.of(fieldName, (float) (boostScore)));
+      fields.add(Pair.of(fieldName, (float) boostScore));
 
       FieldType fieldType = fieldSpec.getSearchableAnnotation().getFieldType();
       if (TYPES_WITH_DELIMITED_SUBFIELD.contains(fieldType)) {
-        fields.add(Pair.of(fieldName + ".delimited", (float) (boostScore * 0.4)));
+        fields.add(Pair.of(fieldName + ".delimited", ((float) boostScore) * 0.4f));
       }
       if (FieldType.URN_PARTIAL.equals(fieldType)) {
-        fields.add(Pair.of(fieldName + ".delimited", (float) (boostScore * 0.4)));
+        fields.add(Pair.of(fieldName + ".delimited", ((float) boostScore) * 0.4f));
       }
     }
 
     return fields;
   }
 
   private static Optional<QueryBuilder> getPrefixQuery(@Nonnull EntitySpec entitySpec, String query) {
-    BoolQueryBuilder finalQuery =  QueryBuilders.boolQuery();
+    BoolQueryBuilder finalQuery = QueryBuilders.boolQuery();
+    if (query.contains("\"")) {
+      finalQuery.should(QueryBuilders.termQuery("urn", query.replaceAll("\"", ""))
+              .boost(Float.parseFloat((String) PRIMARY_URN_SEARCH_PROPERTIES.get("boostScore")) * EXACT_MATCH_BOOST_FACTOR)
```

Comment:
Wait wait so this on occurs if the query contains " " quotations (at least one) and only applies to the _urn_ field?

Is that really what we want? My experience is that exact match on _NAME_ fields is not very good, which may or may not be in the URN at all. 

Also, do users really want to have to put quotes in order to do exact match? If I search for a Table with the exact name I expect to see it - quotes or no quotes.

Reply:
The urn field is not part of any of the models' search specs. It is always treated as an addition added everywhere. This would only be used in a quoted urn, exact match. Below is where we handle the other fields. 
More important for the non-urn field, the quotes become important when considering synonyms, stemming, and other tokenization steps should not be considered. Fulltext search is not the same as a query where exact matches are implicit, both the field and the search terms are manipulated for getting relevant results over exact match.